### ISSUE TITLE
Cannot use with Cashier

### ISSUE THREAD

**Issue by labomatik** [rinvex/laravel-subscriptions#59]
It's not possible to use laravel-subscription with cashier, there is a conflict when adding the trait to the user model.
Any idea how to solve that?

**Comment by Omranic** [comment]
Didn't try both together before, but PRs are always welcome 😉

**Comment by insign** [comment]
I fixed it using the php documentation. It is ugly, but solves the problem in the official way.

```php
use Billable, HasSubscriptions {
      Billable::newSubscription insteadof HasSubscriptions;
      Billable::subscription insteadof HasSubscriptions;
      Billable::subscriptions insteadof HasSubscriptions;
      HasSubscriptions::newSubscription as _newSubscription;
      HasSubscriptions::subscription as _subscription;
      HasSubscriptions::subscriptions as _subscriptions;
   }
```

**Comment by insign** [comment]
@Omranic a PR probably would use new functions names, what you thank about that?

**Comment by insign** [comment]
A made a PR #74 without breaking changes.

**Comment by Omranic** [comment]
This is now fixed in https://github.com/rinvex/laravel-subscriptions/compare/6f70ed89666a...3eedf73b8a68
https://github.com/rinvex/laravel-subscriptions/commit/3eedf73b8a685f9f6096ad7e30bf5a1c4256fbd9

**Comment by Bariskau** [comment]
@Omranic  Still have the same problem.

**Comment by insign** [comment]
Not tested yet. But is this a breaking change?

**Comment by Omranic** [comment]
@Bariskau what problem do you still have, can you please explain in more details..

@insign Yes, function and trait renames.

**Comment by Bariskau** [comment]
@Omranic  Cashier basically just takes care of the stripe payment part. We have to create user limitations or plan information ourselves. That takes a lot of time. This package offers them, it just can't get paid. We already handle this with cashier. For this reason, it support will be good for many people to work with cashier.

If support will not be given, I recommend [this](https://github.com/renoki-co/cashier-register) package to interested people. You can manage the quota through Cashier.

**Comment by Omranic** [comment]
@Bariskau TBH I still didn't get your points, so what's the issue related to this package exactly?

What you explained is correct, and it's mentioned already in the docs, this package doesn't handle payment.

**Comment by insign** [comment]
@Bariskau is right. HasSubscriptions trait from laravel-subscriptions keeps colliding with Billable trait from Cashier.

It's sad that this issue is so old, had many major new version, and it keeps the same function names...

**Comment by Omranic** [comment]
OK, now I see why's the complains! All these issues were already fixed long time ago, but wasn't merged into `master` branch, it was staying in the `develop` and not tagged yet. I've tagged a new release now v6.0.1 which should include all of these fixes! *🚨⚠️ WARNING: BACKWARD INCOMPATIBLE*

**Comment by insign** [comment]
I'm very happy to test this asap. But since it is a break change, shouldn't
you follow semver.org?

Em ter., 14 de dez. de 2021 19:09, Abdelrahman Omran <
***@***.***> escreveu:

**Comment by Omranic** [comment]
Yes, we should be following semver, and the major changes in v6.x was meant to be this change, hence the major release, however I just noticed it wasn't merged into it. So I tagged a new version, and to me it doesn't make sense to tag a new major release to fix that, right?

Anyway, the backward incompatibility issues is because of the function renames that solve the previous Cashier conflicts, nothing else.
